test/integration/reap_logging_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+require 'integration_test'
+require 'socket'
+
+class ReapLoggingTest < Pitchfork::IntegrationTest
+  AFTER_FORK_FILE = 'after_fork'
+  # TODO: This test is slow.
```

Comment:
This test is significantly slower than it's shell counterpart. I suspect it's because we've moved away from a named pipe, and now are relying on multiple writes to a File on disk.

I tried using `File.mkfifo` for the advantages is provides w.r.t blocked reads, but was running into some "timing" issues.

ie. The worker forks, tries to write to the fifo, but because there's no consumer attempting to read, it hangs. On the contrary, if we place that code before letting the worker fork, then the Ruby execution halts. Probably requires a background thread, but doesn't seem necessary.

Thoughts?

Reply:
Woudl be nice to make it faster, but I don't think it's a blocker:

```
ReapLoggingTest#test_reap_worker_logging_messages = 2.03 s = .
```

2s on my machine is fine for an integration test.

The heartbeat one is a bit more concerning:

```
HearbeatTimeoutTest#test_heartbeat_timeout = 12.86 s = .
```

But also acceptable for now. We can always optimize later.